www/docs/operations.mdx

```diff
@@ -0,0 +1,248 @@
+
+### Running Operations
+
+In the example above, the generator function contains a recipe that says "log
+'Hello World' to the console", but it does not actually execute that recipe
+itself. We need something else to actually do that.
+
+The simplest way is with the aptly named
+[`run()`][run] function. Here it's used to execute the example above:
+
+```javascript
+import { run } from 'effection';
+
+run(function* sayHello() {
+  console.log("Hello World!");
+});
+```
+
+This will print `Hello World!` to the console, just as you'd
+expect.
+
+The return value we get from `run` is a `Task`. A `Task` is both an operation
+and a promise, so we can use it with `await` to integrate Effection into
+existing `async/await` code.
+
+```javascript
+import { run } from 'effection';
+
+try {
+await run(function*() {
+  throw new Error('oh no!');
+})
+} catch (error) {
+  console.error(error);
+}
+```
+
+In addition to `run()`, there is the `main()` function which we
+already met in the introduction.  It takes care of a few critical
+things for you like ensuring that the operation it's running is halted
+when the process or browser shuts down, and printing errors to the
+console in case anything goes wrong. If you're writing a program with
+Effection from scratch, you should most likely use `main()` as the
+entry point. `run()` is more bare-bones, and should only be used when
+you need more fine grained control.
+
+Using `main()`, our example looks like this:
+
+```javascript
+import { main } from 'effection';
+
+await main(function*() {
+  throw new Error('oh no!');
+});
+```
+
+Instead of needing to catch the error like we did before, `main()`
+will handle it for us, including printing it to the console.
+
+### Composing Operations
+
+Entry points like `run()` and `main()` are used usually once at the very
+beginning of an Effection program, but the easiest and most common way to
+run an operation is to include it in the body of another operation. To do this,
+we use the `yield*` keyword, which is the Effection equivalent of `await`.
+
+For example, the `sleep()` operation pauses execution for a specified duration
+before resuming. We can call it from any operation using `yield*`;
+
+```javascript
+import { main, sleep } from 'effection';
+
+await main(function*() {
+  yield* sleep(1000);
+  console.log("Hello World!");
+});
+```
+
+Or, it can be embeded in an operation which itself is embedded in another
+operation:
+
+```javascript
+import { main, sleep } from 'effection';
+
+function* makeSlow(value) {
+  yield* sleep(1000);
+  return value;
+}
+
+await main(function*() {
+  let text = yield* makeSlow('Hello World!');
+  console.log(text);
+});
+```
+
+There is no limit to the number of operations that can be nested within
+each other, and in fact, composition is so core to how operations work that
+_every operation in Effection eventually boils down to a combination of
+just three primitive operations_: `action()`, `resource()`, and `suspend()`.
```

Comment:
We should probably mention or link to pages where this is covered

Reply:
I'm going to add a section on "writing operations" that will cover these.